models/gif_cond_exp.h

```diff
@@ -0,0 +1,450 @@
+/*
+ *  gif_cond_exp.h
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef GIF_COND_EXP_H
+#define GIF_COND_EXP_H
+
+#include "config.h"
+
+#ifdef HAVE_GSL
+
+// Includes from nestkernel:
+#include "event.h"
+#include "archiving_node.h"
+#include "ring_buffer.h"
+#include "connection.h"
+
+#include "universal_data_logger.h"
+
+// Includes from gnu gsl:
```

Comment:
GSL includes should come before nestkernel includes, see [C++ Guidelines](https://nest.github.io/nest-simulator/coding_guidelines_c++#order-of-includes) and the code of `iaf_cond_alpha` as reference example.